### DISCUSSION TITLE
BG lvl change in 5 or 10mins

### DISCUSSION THREAD

**Discussion by xamrex** [NightscoutFoundation/xDrip#2745] [Ideas]
Hello
I m using libre2 via bluetooth to get BG readings.
I got problem that sometimes it reads data after 5mins, sometimes after 7 mins, and sometimes after 10mins.

Its cool that it shows difference between BG between actual and previous read.
![UI-indications](https://user-images.githubusercontent.com/12885918/227328855-ffeef343-c7a5-432b-a675-b01fb8ea39b2.png)

But is it possible to add feature that would show how BG is changing every lets say 5mins?

so when i got reading  at 20:00 and its 100mg/dl
and second reading is on 20:08 and its 110 mg/dl
now it will shows 10mg/dl

**is it possible to improve code and add feature that would show that BG change in 5mins is 6.3mg/dl?**
thx

**Comment by Navid200** [comment]
It shows how glucose changes once every 5 minutes.
You can see that if you look at the graph on the main screen.
But, it seems you are not receiving each reading as soon as it is available.  I'm not saying that's not a problem.
But, you definitely have one reading every 5 minutes and every reading is measured exactly at a rate of once every 5 minutes.

**Comment by xamrex** [comment]
It shows how glucose changes once every 5 minutes. -> it shows difference between ACTUAL and previous reading, its described in help
https://xdrip.readthedocs.io/en/latest/use/mainUI/#meet-the-user-interface
**1.How long ago was the reading received and what's the BG difference with the previous value receive**

As i said i m reviving readings randomly, after 5mins, after 7mins, after 9mins etc
And as i know i m not the only one.

Here is a proof:
![Screenshot_20230323_211458_com eveningoutpost dexdrip](https://user-images.githubusercontent.com/12885918/227342785-44aa2f94-28df-4d00-a1f3-909ecb7c2f17.jpg)

![Screenshot_20230323_211502_com eveningoutpost dexdrip](https://user-images.githubusercontent.com/12885918/227342776-d5a8d1d6-08b6-4868-b934-7bccef91ede2.jpg)

**Reply by Navid200** [reply]
What version of xDrip is this?
Please post a screenshot of the classic status page: https://navid200.github.io/xDrip/docs/xDrip-Version.html

**Comment by xamrex** [comment]
2022.11.16
![Screenshot_20230323_212400_com eveningoutpost dexdrip](https://user-images.githubusercontent.com/12885918/227344779-876cc3bd-7158-4386-a6f8-1ab0961368c0.jpg)

**Reply by keencave** [reply]
Are you using a LimiTTer device or do you receive the BG readings via the sensors BT?

**Comment by xamrex** [comment]
I receive BG readings via bluetooth form libre 2.
to be more specific i got  OOP2 app to get readings from libre 2

**Reply by keencave** [reply]
Have you checked the memeory usage of your phone. Full memory can cause certain delays in BT protocol.

**Reply by keencave** [reply]
and please enable loggin with Settings->Less common settings->Extra Logging Settings->extra tags for logging.
If not set please enter: BgReading:d,xdrip libre_receiver:v.
In the three dot menu please go to View Events log and check the timestamps of e.g. entries like BgReading if they show bigger time glitches.

**Comment by xamrex** [comment]
Hey
I am suing 70GB of 128GB, so memory is not an issue.

Here are all pics, and I also inculded data from oop2
![Screenshot_20230324_125730_com eveningoutpost dexdrip](https://user-images.githubusercontent.com/12885918/227516159-5ccc71ba-0a82-426e-bbd9-c96321ea8bb7.jpg)
![Screenshot_20230324_125800_com hg4 oopalgorithm oopalgorithm2](https://user-images.githubusercontent.com/12885918/227516163-6e6053a4-48a2-4679-80e8-f1ac7acc2c72.jpg)
![Screenshot_20230324_125813_com eveningoutpost dexdrip](https://user-images.githubusercontent.com/12885918/227516164-07c03088-97f2-4216-9879-fd1c543210d6.jpg)
![Screenshot_20230324_125821_com eveningoutpost dexdrip](https://user-images.githubusercontent.com/12885918/227516167-912f54ca-1dfc-4f57-bd90-162d1ba5a594.jpg)
![Screenshot_20230324_125828_com eveningoutpost dexdrip](https://user-images.githubusercontent.com/12885918/227516174-4cfbee9c-4ef0-4902-8da6-81c1672de1d7.jpg)
![Screenshot_20230324_125833_com eveningoutpost dexdrip](https://user-images.githubusercontent.com/12885918/227516177-fc1df07b-eeb9-4a3b-b141-e302d36b1173.jpg)
![Screenshot_20230324_125840_com eveningoutpost dexdrip](https://user-images.githubusercontent.com/12885918/227516184-fb8ef1f4-8c19-41ae-a18a-e29c21b67adc.jpg)

thx for help

**Comment by xamrex** [comment]
@keencave any possibility, to add feature in next xdrip+ release to  show how BG is changing every lets say 5mins? (not between actual and previous reading) ?

**Comment by tzachi-dar** [comment]
@xamrex the bg difference shown is the diff per 5 minutes. I'll be extremely surprised if this is not the case. (I'll double check that tomorrow just to be sure).
As for the non constant readings. Can you please add one more tag to the logging? so you should have something like:
Settings->Less common settings->Extra Logging Settings->extra tags for logging.
BgReading:d,xdrip libre_receiver:v,DexCollectionService:d

Then run for a couple of hours. After that, please export your DB and send it tzachi.dar@gmail.com
I'll take a look from where the time differences are coming, but I believe they are coming from BT issues.

**Comment by xamrex** [comment]
Tzachi-dar
You re 100% right (I dont know why I thought that this BG lvl change is diff between actual and previous reading).
Even when readings are not separated by 5mins but longer, it still calculate BG diff in 5 mins ;)

I have read 4 topics on FB group where people with phone P30 lite had issues with Bluetooth readings, so I also think that this time differences are because of BT issues.

Even in OOP2 sometimes timestamps are >10mins

![Screenshot_20230329_085209_com hg4 oopalgorithm oopalgorithm2](https://user-images.githubusercontent.com/12885918/228453748-62b0ef3d-68a0-4810-8be4-482c29b4667e.jpg)
![Screenshot_20230329_090611_com hg4 oopalgorithm oopalgorithm22](https://user-images.githubusercontent.com/12885918/228453750-6c085899-6914-4e2b-acc5-b00ea885f1a7.jpg)

I coudn't find option to export logs to file, so i have created screenshots.
They are available here:
https://photos.app.goo.gl/d2zyzzAwL86ERW6W8

I think we can close this topic, because i finally realized that this BG change is really showing BG diff within 5mins.

**Comment by xamrex** [comment]
@tzachi-dar
So I know why I thought that BG change lvl is not show correctly.
Take a look @ my pics.
Two pics (in app pics) showing the correct value of BG changing within 5 mins.

**Last pic (widget) is not showing the correct value. it shows BG change between current and last reading.**
Can you fix it ?

![2](https://user-images.githubusercontent.com/12885918/228881300-0049a373-3b01-4f65-9112-fddbba50b05a.jpg)
![3](https://user-images.githubusercontent.com/12885918/228881307-3fbb4335-586e-43d6-9777-2a81198153fa.jpg)

![1](https://user-images.githubusercontent.com/12885918/228881326-315b5183-b4e6-410a-aa9b-c7f33060f693.jpg)
